stopcovid/drill_progress/status.py

```diff
@@ -14,15 +20,23 @@ def handle_dialog_event_batches(batches: List[DialogEventBatch]):
             initiator.trigger_first_drill(batch.phone_number, str(batch.batch_id))
 
     user_repo = DrillProgressRepository()
+    state_repo = DynamoDBDialogRepository()
     for batch in batches:
         user_repo.update_user(batch)
-        if initiates_subsequent_drill(batch):
+        if initiates_subsequent_drill(batch, state_repo):
             initiator.trigger_next_drill_for_user(batch.phone_number, str(batch.batch_id))
 
 
 def initiates_first_drill(batch: DialogEventBatch):
     return any(event for event in batch.events if isinstance(event, UserValidated))
 
 
-def initiates_subsequent_drill(batch: DialogEventBatch):
-    return any(event for event in batch.events if isinstance(event, NextDrillRequested))
+def initiates_subsequent_drill(batch: DialogEventBatch, repo: DialogRepository):
```

Comment:
nit: This method should just take in the dialog_state for a user. It doesn't need to know that it comes from a dialog repository

Reply:
good suggestion!